### PR TITLE
rename `sha` to `hash` in variables/methods

### LINKED ISSUE #319: Unify the usage of `sha` vs `hash` in code
I'd vote for `hash`, but that's mostly a matter of taste, both are pretty clear

### PR DESCRIPTION
**Pull request by AABur**
fix #319

### CODE REVIEW

File: git_machete/client.py [hunk 1]
```diff
@@ -1377,8 +1377,8 @@ def is_excluded_reflog_subject(sha_: str, gs_: str) -> bool:
             debug(f"skipping any reflog entry with the hash equal to the hash of the earliest (branch creation) entry: {earliest_sha}")
```
Comment (PawelLipski) [inline git_machete/client.py]: Still quite a few left! Pls check `git grep -n _sha`

Comment (PawelLipski) [inline git_machete/client.py]: Still quite a few left! Pls check `git grep -n _sha`

Comment (AABur) [reply git_machete/client.py]: I fixed and rechecked it. I hope I didn’t miss anything, and no strange names were formed.

**Comment by PawelLipski** [comment]
Just two stray occurrences left:

```
completion/git-machete.completion.zsh:168:  'fork-point:Display SHA of the fork point commit of a branch'
docs/source/cli_help/fork-point.rst:26:Without any option, displays full SHA of the fork point commit for the <branch>.
```

The ones in blogs/ are okay to stay

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**